test/libsolidity/SemanticTest.cpp

```diff
@@ -331,11 +359,28 @@ void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePre
 		return;
 
 	_stream << _linePrefix << "// ====" << endl;
-	if (m_compileViaYulCanBeSet)
+	if (m_compileToEwasmCanBeSet)
+	{
+		// if test was already configured to run via yul,
+		// we need to preserve the original settings.
+		if (m_runWithYul)
+		{
+			// if test was also configured to run without yul.
+			if (m_runWithoutYul)
+				_stream << _linePrefix << "// compileViaYul: also\n";
+			// if test was configured only to run with yul.
+			else
+				_stream << _linePrefix << "// compileViaYul: true\n";
+		}
+
+		_stream << _linePrefix << "// compileToEwasm: also\n";
+	}
+	else if (m_compileViaYulCanBeSet)
```

Comment:
Is the case where both `m_compileViaYulCanBeSet` and `m_compileToEwasmCanBeSet` are true impossible? If so, this needs an assert. If not, then I think this will not set one of the flags properly if you run with `--enforce-via-yul` and `--enforce-compile-to-ewasm`.

Reply:
From my point of view `--enforce-via-yul` and `--enforce-compile-to-ewasm` should be allowed at the same time. I think I fixed the logic regarding the flags.

Reply:
They you either should remove the `else` (if they are truly independent) or add an assert (if one implies the other):
```c++
if (m_canEnableEwasmRun)
{
    soltestAssert(m_canEnableYulRun, "");
    ...
}
```